### ISSUE TITLE
Make it easier to delete local repository copies

### ISSUE THREAD

**Issue by lagartoflojo** [SUSE/rmt#127]
In addition to the "rmt-cli repos disable" or rmt-cli products disable" commands, RMT could have a way of removing repos that are no longer needed from local storage.

Originally from: https://bugzilla.suse.com/show_bug.cgi?id=1087785

**Comment by ikapelyukhin** [comment]
SMT has the following:

```
# smt-repos -h
   ...
   --delete                    : delete the repository from disk
```

**Comment by lagartoflojo** [comment]
Instead of actually implementing a "delete" command, we could also just print out where the repos are on the disk when calling `repos disable` or `products disable`. That gives users the chance to quickly remove the files if they want to.

**Comment by thutterer** [comment]
https://github.com/SUSE/rmt/issues/127#issuecomment-378573900 could be done as part of #94

**Comment by tmuntaner** [comment]
@lagartoflojo I'm not sure if we want to do this with deduplication in place.

If a user deletes a directory that has a package used in symlinks, other products using that symlink could be broken. This would be fixed in the next mirror, but I'm not sure if we should find that "downtime" acceptable.

**Comment by ikapelyukhin** [comment]
@tmuntaner but there are no symlinks? It's only hardlink or copy.

**Comment by tmuntaner** [comment]
@ikapelyukhin You're right, you can ignore my previous comment ^_^

**Comment by NeimadTL** [comment]
Hi guys, I'd like to take this one as @thutterer suggested me last time on [#202](https://github.com/SUSE/rmt/issues/202) . Can somebody tell me where to start please ?

**Comment by ikapelyukhin** [comment]
@NeimadTL At the moment there's no way to remove repositories that are no longer needed (repositories that are disabled for mirroring). Another issue is that repositories accumulate old packages over time: new versions of packages come out, repository metadata references new packages only -- which means that old packages become effectively inaccessible by client machines, but they are still stored on the disk of RMT server.

There are 3 possible things you could do:

1. The quick-and-dirty solution: when repository is being disabled, print the path to it on disk, so that the user can remove it manually. However, during mirroring all file names and checksums are stored in the DB (for creating hardlinks). This solution wouldn't address cleaning up the DB (that's the "dirty" bit).
2. Implement a new `rmt-cli repos clean` command that would go through the directories stored on the disk and remove the ones that aren't mirrored, as well as clean up entries in the DB.
3. Implement a new `rmt-cli repos clean` command that would go through the mirrored repositories and remove files that aren't referenced by the metadata anymore, as well as clean up entries in the DB.

That's the overview of things from my perspective. Please feel free to ask questions, you can also reach me on IRC, RocketChat or via email.

Thanks, Ivan.

**Comment by NeimadTL** [comment]
Hi @ikapelyukhin, thank you for your guidance. I think I'll definitely have a lot more questions so I'll reach out to you one way or another tomorrow. No need to ask them here.

**Comment by tmuntaner** [comment]
Hello @NeimadTL,

 I just wanted to check in with you to see if you're still interested in working on this. If not, I'll create a task for the team to take over.

Thanks

**Comment by NeimadTL** [comment]
I have to admit that I've put on the the side because I wanted to to much things at the same time. I don't how a gem is organize in general and I don't know the RMT project in particular. Is that possible to give me one week or two to deal with it please ? if you have deadlines, I understand.

**Comment by tmuntaner** [comment]
@lagartoflojo Do you think we can close this issue with the latest changes made to RMT?

**Comment by lagartoflojo** [comment]
Solved by #595
